Update interceptors.js

marknokes opened the pull request:
add error handling to fullResponse for simplified troubleshooting

knicola commented:
Thank you for taking time to contribute and I apologize for the delay! I'll be reviewing this in a bit.

marknokes commented:
My pleasure! Thank you for your great work on this. I've been building an automated strategy using your library and it's been really fun!

knicola reviewed (changes requested):
Besides the change requested in the comment above, it is always nice to have a test accompanying any proposed changes to the code for coverage. You can look at this test as an example to get you started: https://github.com/knicola/tdameritradejs/blob/097f5dab7cf72cacfe3a81ebad8e3a377a1586f3/tests/client.spec.js#L903
However, if you are not sure about the test let me know and I can take care of it. Thanks!

knicola commented:
> My pleasure! Thank you for your great work on this. I've been building an automated strategy using your library and it's been really fun!

I'm really glad to hear! It's always nice to get positive feedback. All the best with your strategy!

marknokes commented:
Yes, of course! I'm sorry for the hasty pull request. I have updated for consistency. I wasn't sure about how to perform the test. Please let me know if my latest changes look okay. For the error response when `client.config.returnFullResponse` is set to false, I was attempting to only return what I considered the most significant information (the status and message). I have added the lodash get method to retrieve the values from the error response object.

marknokes commented on src/client/interceptors.js:
```diff
@@ -81,6 +81,11 @@ function fullResponse(client) {
             ? response
             : get(response, 'data')
+    }, error => {
+        return {
```
Yes, of course! I'm sorry for the hasty pull request. I have updated for consistency. I wasn't sure about how to perform the test. Please let me know if my latest changes look okay. For the error response when client.config.returnFullResponse is set to false, I was attempting to only return what I considered the most significant information (the status and message). I have added the lodash get method to retrieve the values from the error response object.

knicola replied:
No worries! I'll write the test over the weekend before merging but for now I believe we're good. Much appreciated!

knicola commented:
Hello, apologies for the long delay. Life got a bit busy.

While writing the tests, I realized that the "error" needs to be thrown rather than returned otherwise it'll be considered as handled and will never trigger `.catch()`. Also, we need to take into account any errors that do not contain a server response (ie. a network error).

I will not be using this PR, as I see you've pushed more commits that are not related to the original issue, but I do have a fix on the way and you will be added as a co-author.

PS. I'll be happy to discuss the additional commits, if that is something you are interested in. Feel free to open a new issue or even a [gh discussion](https://github.com/knicola/tdameritradejs/discussions/categories/ideas).
